internal/token/generate_rsa.go

```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package token
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"encoding/base64"
+	"strconv"
+
+	"github.com/golang-jwt/jwt/v5"
+	"github.com/google/uuid"
+
+	api "github.com/go-vela/server/api/types"
+	"github.com/go-vela/server/database"
+)
+
+// GenerateRSA creates an RSA key pair and sets it in the token manager and saves the JWK in the database.
+func (tm *Manager) GenerateRSA(db database.Interface) error {
+	// generate key pair
+	privateRSAKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		return err
+	}
+
+	// assign KID to key pair
+	kid, err := uuid.NewV7()
+	if err != nil {
+		return err
+	}
+
+	// abstract the JWK from the public key information
+	key := api.JWK{
+		Algorithm: jwt.SigningMethodRS256.Name,
+		Kid:       kid.String(),
+		Use:       "sig",
+		Kty:       "RSA",
+		N:         base64.RawURLEncoding.EncodeToString(privateRSAKey.PublicKey.N.Bytes()),
+		E:         base64.RawURLEncoding.EncodeToString([]byte(strconv.Itoa(privateRSAKey.PublicKey.E))),
```

Comment:
this might have to be 
```
e := new(bytes.Buffer)
binary.Write(e, binary.BigEndian, int64(privateRSAKey.PublicKey.E))
...
E:         base64.RawURLEncoding.EncodeToString(e.Bytes()),
```

Reply:
i think Itoa will give you the ASCII representation which might need additional unmarshaling code